### PR TITLE
Add HPX

### PR DESCRIPTION
**Pull request by havogt**

### CODE REVIEW

File: gcc-hpx/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+ARG REPOSITORY
+ARG GCC_VERSION=9
+FROM ${REPOSITORY}:gcc-${GCC_VERSION}
+LABEL maintainer="Felix Thaler <thaler@cscs.ch>"
+
+ARG HPX_TAG=master
+ENV HPX_TAG ${HPX_TAG}
+
+RUN rm -rf /usr/local/include/boost && \
```
Comment (havogt) [inline gcc-hpx/Dockerfile:9]: maybe we should add boost with libs in the base, but I thought for now we can do it here to keep the other images smaller.

File: build.sh [hunk 2]
```diff
@@ -31,4 +31,10 @@ done
 for clang_version in 10; do
     docker build --build-arg REPOSITORY=$1 --build-arg CLANG_VERSION=$clang_version -t $1:clang-$clang_version-hip clang-hip
     docker build --build-arg REPOSITORY=$1 --build-arg BASE=clang-$clang_version-hip -t $1:test-clang-$clang_version-hip test
-done
\ No newline at end of file
+done
+
+# GCC with HPX (not by default as it will increase docker build time significantly)
```
Comment (havogt) [inline build.sh:36]: Long term, e.g. once there is a release that is working for us, we can add it to the normal gcc image.

File: gcc-hpx/Dockerfile (hunk 1 above)
Comment (fthaler) [reply gcc-hpx/Dockerfile:9]: If the system boost version is good enough, that’s probably the most elegant solution. IIRC, when setting up the containers, the available system boost was too old for GT. But I don’t care much if we do it now or later…

File: gcc-hpx/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+ARG REPOSITORY
+ARG GCC_VERSION=9
+FROM ${REPOSITORY}:gcc-${GCC_VERSION}
+LABEL maintainer="Felix Thaler <thaler@cscs.ch>"
+
+ARG HPX_TAG=master
+ENV HPX_TAG ${HPX_TAG}
```
Comment (fthaler) [inline gcc-hpx/Dockerfile:7]: Why do we need HPX_TAG at runtime?

File: gcc-hpx/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,21 @@
+ARG REPOSITORY
+ARG GCC_VERSION=9
+FROM ${REPOSITORY}:gcc-${GCC_VERSION}
+LABEL maintainer="Felix Thaler <thaler@cscs.ch>"
+
+ARG HPX_TAG=master
```
Comment (fthaler) [inline gcc-hpx/Dockerfile:6]: Better to move those lines to further below (just before the installation of HPX), then changing the HPX version does not trigger a re-installation of boost.

**Pull request merged**